### PR TITLE
PR - Frontend - Driftquiz (Quiz App)

### PR DESCRIPTION
**Pull request by duhbhavesh**
#### Please review Driftquiz (Quiz App), and suggest necessary feedbacks.
 
_Thank you._

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { FC } from 'react';
+import { Header } from './components/Header';
+import { Routes, Route } from 'react-router-dom';
+import { Home } from './pages/Home';
+import { Quiz } from './pages/Quiz';
+import { QuizReport } from './pages/QuizReport';
+
+export const App: FC = () => {
```
Comment (singhadi672) [inline src/App.tsx]: is there any particular reason to use this?

File: src/reducers/QuizReducer.types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { ResultType } from '../context/QuizContext.types';
+import { QuizDataType, QuizType } from '../data/QuizData.types';
+
+export type ActionType =
```
Comment (singhadi672) [inline src/reducers/QuizReducer.types.ts:4]: i liked how you have extracted the types too

File: src/reducers/QuizReducer.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+         const currentQuestion = state.currentQuestionNumber + 1;
+         return {
+            ...state,
+            currentQuestionNumber: currentQuestion,
+         };
+
+      case 'INCREASE_SCORE':
+         return { ...state, score: state.score + action.payload };
+
+      case 'RESET':
```
Comment (singhadi672) [inline src/reducers/QuizReducer.tsx]: i think you can remove the reset state aa the default is also returning the same state?

File: src/pages/Quiz.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { FC, useEffect } from 'react';
+import { useParams } from 'react-router';
+import { Question } from '../components/Question';
+import { QuizHeader } from '../components/QuizHeader';
+import { useQuiz } from '../context/QuizContext';
+
+export const Quiz: FC = () => {
+   const { state, dispatch } = useQuiz();
```
Comment (singhadi672) [inline src/pages/Quiz.tsx:8]: here also i think you can eliminate explicitly giving React.fc

File: src/components/Question.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+type QuizProp = {
+   currentQuiz: QuizType;
+};
+
+export const Question = ({ currentQuiz }: QuizProp) => {
+   const navigate = useNavigate();
+   const { state, dispatch } = useQuiz();
+   const [disabledButtons, setDisabledButtons] = useState<boolean>(false);
+   const [optionID, setOptionID] = useState<string>('');
```
Comment (singhadi672) [inline src/components/Question.tsx:14]: i liked the way you took care of the useState types too

**Review by singhadi672 — COMMENTED** [review]
Hi @duhbhavesh ,  clean code and the UI also looks great!!

File: src/App.tsx (hunk 1 above)
Comment (duhbhavesh) [reply src/App.tsx]: As far as I know, the return value is valid JSX.

**Comment by singhadi672** [comment]
hi @duhbhavesh the changes are good to go!

**Review by barbaricankit — COMMENTED** [review]
Nice job man

**Pull request merged**